src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -872,11 +873,11 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
     }
 
     pub fn rustflags_args(&self, unit: &Unit) -> CargoResult<Vec<String>> {
-        env_args(self.config, &self.build_config, unit.kind, "RUSTFLAGS")
+        env_args(self.config, &self.build_config, &self.target_info, unit.kind, "RUSTFLAGS")
```

Comment:
I think this may want to switch between `target_info` and `host_info` based off `unit.kind`